clusterloader2/pkg/test/simple_test_executor.go

```diff
@@ -318,6 +320,52 @@ func (ste *simpleTestExecutor) ExecuteObject(ctx Context, object *api.Object, na
 	return errList
 }
 
+// initNamespaceParamters initializes namespace parameters
+func initNamespaceParameters(ctx Context, conf *api.Config) *errors.ErrorList {
+
+	// For backwarad compatibility with deprecated "automananaagedNamespaces" in testconfig file.
+	// Remove in the future versions
+	if conf.Namespace.Count == nil && conf.AutomanagedNamespaces != nil {
+		conf.Namespace.Count = conf.AutomanagedNamespaces
+	}
+
+	var defaultNamespaces int32 = 1
+	if conf.Namespace.Count == nil {
+		conf.Namespace.Count = &defaultNamespaces
+	}
+	if *conf.Namespace.Count <= 0 {
+		return errors.NewErrorList(fmt.Errorf("number of namespaecs: %d was less than 1", *conf.Namespace.Count))
+	}
```

Comment:
This technically has nothing to do with executor.
So far we never have a need for it, but I would prefer to:
- in api/ directory introduce the default.go file
- in that file define functions like:
```
func (c *Config) SetDefaults() {
  c.Namespace.SetDefaults()
}

func (n *NamespaceConfig) SetDefaults() {
  // defaulting logic that you have here
}
```

On top of that, introduce a `validation.go` where you would introduce define:
```
func (c *Config) Validate() errors.ErrorList {
  return c.Namespace.Validate()
}

func (n *NamespaceConfig) Validate() errors.ErrorList {
  // return what you have here
}
```

And call defaulting and validation here:
https://github.com/kubernetes/perf-tests/blob/d9bdf46ed8414e87b3c50d49f14a64e9e8970051/clusterloader2/pkg/test/test.go#L63

Reply:
> This technically has nothing to do with executor.
> So far we never have a need for it, but I would prefer to:
> 
> * in api/ directory introduce the default.go file
> * in that file define functions like:
> 
> ```
> func (c *Config) SetDefaults() {
>   c.Namespace.SetDefaults()
> }
> 
> func (n *NamespaceConfig) SetDefaults() {
>   // defaulting logic that you have here
> }
> ```
> 
> On top of that, introduce a `validation.go` where you would introduce define:
> 
> ```
> func (c *Config) Validate() errors.ErrorList {
>   return c.Namespace.Validate()
> }
> 
> func (n *NamespaceConfig) Validate() errors.ErrorList {
>   // return what you have here
> }
> ```
> 
This would certainly lead to a neater implementations, but I have two comments/questions. 

1. Supporting the deprecated command line options _delete-stale-namespaces_ would be difficult because SetDefaults does not have access to config.ClusterConfig.DeleteStaleNamespaces. I'd suggest we just delete this command flag.

2.  To continue support automanagedNamespaces in the testconfig file would be tricky too. Without using *int32, when Count's value is 1 (the default), it's impossible to tell if it's explicitly set or the default value is used. In practice, a user should only specify the number in one or the other way, but not both.  In case both are specified (not equal to 1), we'll use the Count's value.
```
@wojtek-t

Reply:
done